packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1310,6 +1324,15 @@ class WidgetsApp extends StatefulWidget {
     VoidCallbackIntent: VoidCallbackAction(),
   };
 
+  /// The default value for [onNavigationNotification].
+  ///
+  /// Updates the platform with [NavigationNotification.canHandlePop] and stops
+  /// bubbling.
+  static bool defaultOnNavigationNotification(NavigationNotification notification) {
```

Comment:
Oh, that is a good question: What happens if there are multiple sibling navigators?

Reply:
See the example [pop_scope.1.dart](https://github.com/flutter/flutter/pull/120385/files#diff-f860e82bd00ff2e5ef01e93a7c50b1ab3a93aaf88d232203b1bc85992180166d).  The developer has to handle back gestures for each nested sibling Navigator with a PopScope widget, but then it works.

Reply:
pop_scope.1.dart doesn't keep the state of navigator when switch tabs? In this case, there will only be one navigator in the widget tree at a time, so it is not sibling navigator scenario I think?

Reply:
It does keep track of the state of each tab, but it does so manually via the Navigator.pages API.  I guess there are only ever 2 Navigators in the widget tree at a time (one in the MaterialApp and one in the current tab) assuming that the old tab is immediately disposed when switching to a new tab.

Do you have an example of the kind of sibling set up that you want to see?

Reply:
I tried the example in the CupertinoTabScaffold docs and found that it never supported system back gestures.  On master a back gesture just closed the app regardless of routes on either tab.

But I made some changes to CupertinoTabView to get it to work in 814e202be23060fd137abcd51052a50b2270e531.  The main takeaway is that for sibling Navigators, it's up to the developer to handle enabling the active sibling and disabling the inactive sibling(s), but it works if you do that.

Some caveats to think about from that commit:

  1. I added a `canPop` override parameter to NavigatorPopHandler.  This let me effectively switch it on/off easily based on the active sibling.  I couldn't just remove it from the widget tree to disable it because then it would miss notifications from its Navigator (a problem brought up by @goderbauer in https://github.com/flutter/flutter/pull/120385#discussion_r1272725300).
  2. I determine if a tab is active using TickerMode.of, which is sort of relying on an [implementation detail](https://github.com/flutter/flutter/blob/fb7a763c640d247d090cbb373e4b3a0459ac171b/packages/flutter/lib/src/cupertino/tab_scaffold.dart#L486-L492) of CupertinoTabScaffold.  Is there a better way?
  3. Do we even care to support this?  It didn't work before, and I think it would only matter when using CupertinoTabScaffold on Android, which I imagine is uncommon.